routes.py

```diff
@@ -18,6 +18,8 @@
         handlers.api.UserHandler),
     (r"/api/(user)", handlers.api.UserHandler),
     (r"/api/shakes", handlers.api.UserShakesHandler),
+    (r"/api/(shake_name|shake_id)/([a-zA-Z0-9_\-]+)",
```

Comment:
I just copied the regex from the user_name route. Now that I look closer, I see that the other Shake routes have a different regex. I assume it should match user_name since this route would accept either a numeric ID or a path, which could include underscores or hyphens?

Reply:
I'd probably declare this as two routes, since the pattern for `shake_id` is one thing and the pattern for `shake_name` is another (ideally the user route would be done this way, as well).

    (r"/api/shake_id/([0-9]+)",
        handlers.api.ShakeHandler),
    (r"/api/shake_name/([^a-zA-Z0-9\-]+)",
        handlers.api.ShakeHandler),

Note the singular `ShakeHandler` which matches `UserHandler` since the handler is only returning a single object.

I'd also amend the handler so it returns a proper 404 error if the lookup fails.